### PR TITLE
Initial commit for core-edge

### PR DESCRIPTION
**Pull request by mneedham**
## catchup

 Store copy and transaction pulling, similar to the way slaves catch up with
 master in the traditional HA setup.
## discovery

Uses Hazelcast to discover an initial cluster topology both for Core and Edge
instances. This discovery layer is used to prompt RAFT cluster changes in Core as
instances leave and join, and used as a topology manager for Edge instances.
## raft

An implementation of a distributed log kept consistent by RAFT. While there
are several packages in here, the key ones are:
- log - where the log implementations for the RAFT protocol are kept
- replication - machinery for replicating entries in the raft log
- membership - where RAFT group membership is maintained (see above for
  initial setup)
## server

 The server package is where all the code is brought together. It also provides
 some shared infrastructure for sending messages, dealing with settings etc.

### CODE REVIEW

**Comment by lutovich** [comment]
Holy shit! This is impossible to review! Can it be at least split into multiple commits?

**Comment by davidegrohmann** [comment]
It would have been nice to have preserved history for future reference. What is the reason for having squashed everything into a single commit?

File: community/kernel/pom.xml [hunk 1]
```diff
@@ -134,7 +134,7 @@ the relevant Commercial Agreement.
         <configuration>
           <dotGitDirectory>${git.directory}</dotGitDirectory>
           <failOnNoGitDirectory>false</failOnNoGitDirectory>
-          <useNativeGit>true</useNativeGit>
+          <useNativeGit>false</useNativeGit>
```
Comment (davidegrohmann) [inline community/kernel/pom.xml]: This is actually a very good example of a change that should have been in a separate commit since there is no way for me to understand why this change has been made.

Comment (chrisvest) [inline community/kernel/pom.xml]: I think it caused problems for @jimwebber, for some reason, so @digitalstain said he'd make this change and see if it caused any problems. Apparently it hasn't for anyone in the core-edge team, since it's still there.

Comment (davidegrohmann) [inline community/kernel/pom.xml]: I am a bit lost here, let me understand. Are you saying that this is actually a mistake?

**Comment by srbaker** [comment]
A single, 10,000 line commit? How did you work for so long without committing and not losing anything?

File: community/kernel/pom.xml (hunk 1 above)
Comment (martinfurmanski) [inline community/kernel/pom.xml]: This will use the embedded jgit instead of whatever git happens to be available in your environment.

"As rule of thumb - stay on `jgit` (keep this `false`) until you notice performance problems."

Comment (davidegrohmann) [inline community/kernel/pom.xml]: Right, it sounds reasonable. Can we please have this change in its own PR?
This is completely unrelated to core-edge.

Comment (mneedham) [inline community/kernel/pom.xml]: @davidegrohmann here we go: https://github.com/neo4j/neo4j/pull/5948

Comment (davidegrohmann) [inline community/kernel/pom.xml]: @mneedham thanks, shall we remove then that change from this PR?

Comment (mneedham) [inline community/kernel/pom.xml]: @davidegrohmann good idea. Done.
